apps/st2-history/history-popup.component.js

```diff
@@ -60,7 +63,21 @@ export default class HistoryPopup extends React.Component {
 
   handleSubmit(e) {
     e.preventDefault();
-
+    const hasValue = Object.values(this.state.payload).includes('********');
+    let payLoadKey;
+    if (hasValue === true) {
+      payLoadKey =  Object.keys(this.state.payload).find(key => this.state.payload[key] === '********');
+    }
+
+    const isValue = Object.values(this.state.payloadCopy).includes('********');
+    let payloadCopyKey ;
+    if (isValue === true) {
+      payloadCopyKey =  Object.keys(this.state.payloadCopy).find(key => this.state.payloadCopy[key] === '********');
+    }
+    
+    if (this.state.payload[payLoadKey] === this.state.payloadCopy[payloadCopyKey]) {
+      delete this.state.payload[payLoadKey];
+    }
```

Comment:
Can you add comments on how this section of changes work? Which dictionary (payLoadKey or payLoadCopyKey) is the one being modified when I override the secret input parameter in the UI?